py/desispec/scripts/zproc.py

```diff
@@ -116,26 +130,40 @@ def main(args=None, comm=None):
     if not isinstance(args, argparse.Namespace):
         args = parse(options=args)
 
+    if args.starttime is not None:
+        start_time = args.starttime
+    else:
+        start_time = time.time()
+
     log = get_logger()
 
-    # if args.healpix is not None:
-    #     msg = f"Healpix redshifts is not yet implemented"
-    #     log.error(msg)
-    #     raise NotImplementedError(msg)
-    if args.tileid is None:
-        msg = "Must specify tile"
-        log.error(msg)
-        raise ValueError(msg)
+    #- set default groupname if needed (cumulative for tiles, otherwise healpix)
+    if args.groupname is None:
+        if args.tileid is not None:
+            args.groupname = 'cumulative'
+        elif args.healpix is not None:
+            args.groupname = 'healpix'
+        else:
+            msg = 'Must specify --tileid or --healpix'
+            log.critical(msg)
+            raise ValueError(msg)
+
+    #- consistency of options
+    if args.groupname == 'healpix':
+        assert args.healpix is not None, "--groupname healpix requires setting --healpix too"
+        assert args.nights is None, f"--groupname healpix doesn't use --nights {args.nights}"
+        assert args.expids is None, f"--groupname healpix doesn't use --expids {args.expids}"
+        assert args.thrunight is None, f"--groupname healpix doesn't use --thrunight {args.thrunight}"
+    else:
+        assert args.tileid is not None, f"--groupname {args.groupname} requires setting --tileid too"
 
-    today = int(time.strftime('%Y%m%d'))
     if args.thrunight is not None:
-        ## change if implementing healpix
-        if args.groupname not in ['cumulative']:
+        if args.groupname not in ['cumulative',]:
             msg = f"--thrunight only valid for cumulative redshifts."
             log.error(msg)
             raise ValueError(msg)
-        elif args.thrunight < 20200214 or args.thrunight > today:
-            msg = f"--thrunight must be between 20200214 and today."
+        elif args.thrunight < 20200214:
```

Comment:
I removed it because of a general principle to not forbid simulating + processing the future.  e.g. earlier versions of astropy couldn't calculate the moon altitude above the horizon for the DESI survey simulations because it couldn't look up future IERS coefficients to determine the moon's location to sub-arcsecond accuracy (they've fixed that, but it drove me crazy at the time).  But you make a good point about not implying including data that may not yet exist, so I'll put this check back.  If we need to simulate+process the future (e.g. DESI2...) we can ad an override option.

Reply:
I now understand the philosophical point, and it's true if we run a bunch of data simulations of future data this will break. I like your solution of an override flag that can be set to ignore this constraint. I also agree we should just add back this sanity check now and add the override flag in the future when/if such simulations happen (i.e. not in this PR).